timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/NamedMinTimestampLockCollection.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.lock;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import java.util.Optional;
+
+final class NamedMinTimestampLockCollection {
+    private static final String IMMUTABLE_TIMESTAMP_NAME = "ImmutableTimestamp";
```

Comment:
Now the million dollar question: what would happen if we just changed the naming convention here?

My absolute preference would be to basically start these now to be something like: "nmt:<lock-name>:<timestamp>", because you could very quickly check the prefix to validate everything, and this class could just have a single method "getPrefix() -> String" that LockManager could call to reject anyone that's trying to acquired a named min lease timestamp through ExclusiveLocks.

Reply:
Tracking the possibility of changing the naming convention as a FLUP